Documentation/devel/architecture.md

```diff
@@ -102,3 +101,63 @@ This will activate all the reaper services when one of the targets is activated,
 ### Stage 2
 
 The final stage, stage2, is the actual environment in which the applications run, as launched by stage1.
+
+## Process execution chain
+
+After calling `rkt` the execution chain follows the numbering of stages, having the following general order:
+
+![execution-flow](execution-flow.png)
+
+1. invoking process -> stage0:
+The invoking process uses its own mechanism to invoke the rkt binary (stage0). When started via a regular shell or a supervisor, stage0 is usually forked and exec'ed becoming a child process of the invoking shell or supervisor.
+
+2. stage0 -> stage1:
+An ordinary `exec(3)` is being used to replace the stage0 process with the stage1 entrypoint. The entrypoint is referenced by the `coreos.com/rkt/stage1/run` annotation in the stage1 image manifest.
+
```

Comment:
Good point, actually for the stop execution chain I would prepare a different diagrams, in order not to bloat the startup execution chain diagram.s